website/server/controllers/api-v3/chat.js

```diff
@@ -175,7 +175,8 @@ api.postChat = {
       throw new NotAuthorized(res.t('messageGroupChatSpam'));
     }
 
-    const newChatMessage = group.sendChat(req.body.message, user);
+    const client = req.headers['x-client'].replace('habitica-', '');
```

Comment:
this will throw an error if the `x-client` header isn't set

Reply:
for example if the message is sent from a 3rd party client

Reply:
Good point. Originally didn't have the replace part and then forgot to handle that case. I now added something for that and also set a client for when no header was passed.